scrapers_next/wv/committees.py

```diff
@@ -0,0 +1,151 @@
+from re import sub
+from spatula import HtmlPage, HtmlListPage, CSS, URL, Source, SkipItem
+from openstates.models import ScrapeCommittee
+
+
+class CommitteeMember(HtmlPage):
+    """
+    Scrape a committee member profile, if existing.
+
+    Position is scraped from the directory page and passed directly to this class.
+
+    If a member does not have a profile or their name cannot be gleaned using a
+    css selector, use fallback_name.
+
+    :param position: Rank of the member in the committee currently being scraped.
+    e.g. "Chair"
+    :param fallback_name: The name to use if the legislator's full name cannot be
+    determined from their profile. e.g. "Senator Burns"
+    """
+
+    position: str
+    fallback_name: str
+
+    def __init__(self, input_val, source: Source, position: str, fallback_name: str):
+        super(CommitteeMember, self).__init__(input_val, source=source)
+        self.position = position
+        self.fallback_name = fallback_name
+
+    def process_page(self):
+        # min_items set to zero because not all members have a profile
+        # page instead shows a list of media releases
+        name_match = CSS("#wrapleftcolr > h2:nth-child(1)", min_items=0).match(
+            self.root
+        )
+        name = name_match[0].text if len(name_match) == 1 else self.fallback_name
+        self.input.add_source(self.source.url, f"{name} url")
+        self.input.add_member(sub(r"\(.*\)", "", name), self.position)
+        return self.input
+
+    def get_source_from_input(self):
+        pass
+
+
+class CommitteeDetail(HtmlListPage):
+    """
+    Scrape a list of committee members.
+
+    :param selector: CSS matcher for each legislator url, according to chamber.
+    :param url_signifier: Part of each matched element's href that should exist
+    in a valid member profile url.
+    """
+
+    selector: CSS
+    url_signifier: str
+
+    def __init__(
+        self,
+        input_val: ScrapeCommittee,
+        source: Source,
+        selector: CSS,
+        url_signifier: str,
+    ):
+        super(CommitteeDetail, self).__init__(input_val, source=source)
+        self.selector = selector
+        self.url_signifier = url_signifier
+
+    def process_item(self, item):
+        if self.url_signifier in item.get("href"):
+            next_elem = item.getnext().text
+            position = next_elem if isinstance(next_elem, str) else "member"
```

Comment:
Note: Joint committee directories list positions outside of HTML elements (flat text is a direct child of the enclosing `div`). Thus currently this scraper lists all Joint members with position "member". All other chambers are working as expected.

Reply:
One way to address this different case for Joint Committees could be to set a conditional like `if self.input.chamber == "legislature"` and then you can get access to all the text that exists in the `<td>` elems by calling `.getparent().text_content()` on `item`. This would allow you to parse through the text and get those leadership titles that exist as 'floating text' outside of individual elements.

Then using a regex pattern you can `re.search()` (and then get data in `.groups()`) for where the fallback name is followed by 'whitespace dash whitespace' (i.e. assign the fallback name to a variable earlier in `process_item()`
- so you could make a pattern like `rf"({fallback_name})\s+-\s+(.+)"`
   - using that pattern in a `re.search()` on the text content string for all of a particular chambers members for that commit, you can check if there is a match. If there is, that means you have a leader, and then you can call `.groups()` on that match object and then index to grab the second group (i.e. the characters contained at the end of the regex pattern: `(.+)`.
   - So for [this committee](https://www.wvlegislature.gov/committees/interims/committee.cfm?abb=agri), you'd get a match for that pattern on Senator Sypolt (and can extract "Chair"), but no match for Senator Baldwin (and can just assign "Member".)

Reply:
Brilliant! Thanks for the tips here - worked like a charm. Added conditional parsing in [200d22b](https://github.com/openstates/openstates-scrapers/pull/4398/commits/200d22b22f12d74c88e8a7685520b33448ce6362)